include/crud-posts.php

```diff
@@ -340,8 +347,111 @@ public function force_tags_translation( $post_id, $post_after, $post_before ) {
 		$post_before = $post_before->to_array();
 
 		if ( ! empty( $post_before['tags_input'] ) ) {
-			// Let's ensure that  `PLL_CRUD_Posts::set_object_terms()` will do its job.
+			// Let's ensure that `PLL_CRUD_Posts::set_object_terms()` will do its job.
 			wp_set_post_tags( $post_id, $post_before['tags_input'] );
 		}
 	}
+
+	/**
+	 * Makes sure that all terms in the given list are in the given language.
+	 * If not the case, the terms are translated or created (for a hierarchical taxonomy, terms are created recursively).
+	 *
+	 * @since 3.5
+	 *
+	 * @param WP_Term[]    $terms    List of terms to translate.
+	 * @param string       $taxonomy The terms' taxonomy.
+	 * @param PLL_Language $language The language to translate the terms into.
+	 * @return int[] List of `term_id`s.
+	 *
+	 * @phpstan-return array<positive-int>
+	 */
+	private function translate_terms( array $terms, string $taxonomy, PLL_Language $language ): array {
+		$term_ids_translated = array();
+
+		foreach ( $terms as $term ) {
+			$term_ids_translated[] = $this->translate_term( $term, $taxonomy, $language );
+		}
+
+		return array_filter( $term_ids_translated );
+	}
+
+	/**
+	 * Translates the given term into the given language.
+	 * If the translation doesn't exist, it is created (for a hierarchical taxonomy, terms are created recursively).
+	 *
+	 * @since 3.5
+	 *
+	 * @param WP_Term      $term     The term to translate.
+	 * @param string       $taxonomy The term's taxonomy.
+	 * @param PLL_Language $language The language to translate the term into.
+	 * @return int
```

Comment:
Maybe you could add doc for `@return`?

Reply:
Done in [8d687bf](https://github.com/polylang/polylang/pull/1336/commits/8d687bfbedef0d65af4cae96d8bf64b16c583bab).